src/workload_api/client.rs

```diff
@@ -117,30 +117,81 @@ pub enum ClientError {
     #[error("trust domain in bundles response is invalid")]
     InvalidTrustDomain(#[from] SpiffeIdError),
 
-    /// Error returned by the GRPC library, when there is an error connecting to the Workload API.
-    #[error("error response from the Workload API: {0}")]
-    Grpc(#[from] grpcio::Error),
+    /// Error returned by the GRPC library, when there is an error response from the Workload API.
+    #[error("error response from the Workload API")]
+    Grpc(#[from] tonic::Status),
+
+    /// Error returned by the GRPC library when there is an error creating a transport channel to the Workload API.
+    #[error("error creating transport")]
+    Transport(#[from] tonic::transport::Error),
+}
+
+/// This type represents a client to interact with the Workload API.
+///
+/// Supports one-shot calls and streaming updates for X.509 and JWT SVIDs and bundles.
+/// The client can be used to fetch the current SVIDs and bundles, as well as to
+/// subscribe for updates whenever the SVIDs or bundles change.
+#[allow(missing_debug_implementations)]
+pub struct WorkloadApiClient {
```

Comment:
Another thing, I think this struct needs the `#[derive(Clone)]` since the mutable reference will need to be cloned?

Reply:
Makes sense.